src/main/java/nexus/components/Parser.java

```diff
@@ -0,0 +1,175 @@
+package nexus.components;
+
+import nexus.exceptions.InvalidInputException;
+import nexus.task.Deadline;
+import nexus.task.Event;
+import nexus.task.Task;
+import nexus.task.TaskList;
+import nexus.task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    private static void parseMark(TaskList list, String[] data, Storage storage, Ui ui) throws InvalidInputException {
```

Comment:
Nit: Just a small note that you have exceeded the soft limit (110 characters) on some lines.